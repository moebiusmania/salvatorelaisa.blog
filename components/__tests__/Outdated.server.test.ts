import { describe, it, expect, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import Outdated from "../Outdated.server.vue";

describe("Outdated", () => {
  // Mock current date to ensure consistent testing
  const currentYear = 2024;
  beforeEach(() => {
    vi.useFakeTimers();
    vi.setSystemTime(new Date(currentYear, 0, 1));
  });

  afterEach(() => {
    vi.useRealTimers();
  });

  // Snapshot test
  it("renders correctly when post is outdated", () => {
    const wrapper = mount(Outdated, {
      props: {
        date: "2020-01-01",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("shows warning when post is older than 2 years", () => {
    const wrapper = mount(Outdated, {
      props: {
        date: "2020-01-01",
      },
    });

    expect(wrapper.isVisible()).toBe(true);
    expect(wrapper.text()).toContain("ATTENZIONE!");
    expect(wrapper.text()).toContain(
      "Questo post ha qualche anno sulle spalle"
    );
  });

  it("doesn't show warning for recent posts", () => {
    const wrapper = mount(Outdated, {
      props: {
        date: "2023-01-01",
      },
    });

    expect(wrapper.find("div").exists()).toBe(false);
  });

  it("handles edge case of exactly 2 years", () => {
    const wrapper = mount(Outdated, {
      props: {
        date: "2022-01-01",
      },
    });

    expect(wrapper.find("div").exists()).toBe(false);
  });

  it("renders warning icon", () => {
    const wrapper = mount(Outdated, {
      props: {
        date: "2020-01-01",
      },
    });

    const svg = wrapper.find("svg");
    expect(svg.exists()).toBe(true);
    expect(svg.attributes("xmlns")).toBe("http://www.w3.org/2000/svg");
    expect(svg.attributes("viewBox")).toBe("0 0 24 24");
  });

  it("applies correct styling", () => {
    const wrapper = mount(Outdated, {
      props: {
        date: "2020-01-01",
      },
    });

    const div = wrapper.find("div");
    expect(div.exists()).toBe(true);
    // Add any specific style checks based on your CSS
  });

  it("handles invalid date gracefully", () => {
    const wrapper = mount(Outdated, {
      props: {
        date: "invalid-date",
      },
    });

    expect(wrapper.find("div").exists()).toBe(false);
  });

  it("handles empty date prop", () => {
    const wrapper = mount(Outdated, {
      props: {
        date: "",
      },
    });

    expect(wrapper.find("div").exists()).toBe(false);
  });

  it("calculates year difference correctly", () => {
    // Test various years
    const testCases = [
      { date: "2021-01-01", shouldShow: true },
      { date: "2022-01-01", shouldShow: false },
      { date: "2023-01-01", shouldShow: false },
      { date: "2024-01-01", shouldShow: false },
      { date: "2019-12-31", shouldShow: true },
    ];

    testCases.forEach(({ date, shouldShow }) => {
      const wrapper = mount(Outdated, {
        props: { date },
      });
      expect(wrapper.find("div").exists()).toBe(shouldShow);
    });
  });
});
