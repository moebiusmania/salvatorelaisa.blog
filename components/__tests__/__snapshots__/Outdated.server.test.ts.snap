// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Outdated > renders correctly when post is outdated 1`] = `
"<div data-v-0f20dc1b=""><svg data-v-0f20dc1b="" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24">
    <path data-v-0f20dc1b="" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z"></path>
  </svg><span data-v-0f20dc1b=""><strong data-v-0f20dc1b="">ATTENZIONE!</strong> Questo post ha qualche anno sulle spalle, quindi Ã¨ molto probabile che il contesto degli argomenti trattati non sia piÃ¹ attuale. Controlla la data di pubblicazione prima di continuare a leggere ðŸ™‚</span></div>"
`;
